optuna/samplers/_cmaes.py

```diff
@@ -429,9 +484,22 @@ def sample_relative(
         study._storage.set_trial_system_attr(
             trial._trial_id, generation_attr_key, optimizer.generation
         )
+        study._storage.set_trial_system_attr(
+            trial._trial_id, self._attr_keys.popsize, optimizer.population_size
+        )
```

Comment:
I am wondering if the following code will work correctly without saving a `restart_strategy` in `system_attrs`.

```python
import math
import optuna


def objective(trial: optuna.Trial) -> float:
    return sum([
        math.sin(trial.suggest_float('param-{}'.format(i), -10 * i, 10 * i))
        for i in range(10)
    ])


if __name__ == '__main__':
    storage = optuna.storages.InMemoryStorage()

    ipop = optuna.samplers.CmaEsSampler(restart_strategy="ipop")
    study = optuna.create_study(sampler=ipop, storage=storage, study_name="bipop-example")
    study.optimize(objective, n_trials=100)

    bipop = optuna.samplers.CmaEsSampler(restart_strategy="bipop")
    study = optuna.load_study(sampler=bipop, storage=storage, study_name="bipop-example")
    study.optimize(objective, n_trials=100)

    print(study.best_trial)
```

More precisely, when the restart strategy changes from ipop to bipop, do you think the parameters stored in system_attr such as n_restarts and popsize should be reused?

Reply:
I update code to deal with this problem. Could you take another look?